storage/conformance_test.go

```diff
@@ -16,21 +16,337 @@ package storage
 
 import (
 	"bytes"
+	"context"
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
 	"net/url"
+	"os"
 	"strconv"
 	"strings"
 	"testing"
 	"time"
 
+	"cloud.google.com/go/internal/uid"
 	storage_v1_tests "cloud.google.com/go/storage/internal/test/conformance"
 	"github.com/golang/protobuf/jsonpb"
 	"github.com/google/go-cmp/cmp"
+	"google.golang.org/api/option"
+	raw "google.golang.org/api/storage/v1"
+	htransport "google.golang.org/api/transport/http"
 )
 
+var (
+	bucketIDs           = uid.NewSpace("bucket", nil)
+	objectIDs           = uid.NewSpace("object", nil)
+	notificationIDs     = uid.NewSpace("notification", nil)
+	projectID           = "my-project-id"
+	serviceAccountEmail = "my-sevice-account@my-project-id.iam.gserviceaccount.com"
+)
+
+// Holds the resources for a particular test case. Only the necessary fields will
+// be populated; others will be nil.
+type resources struct {
+	bucket       *BucketAttrs
+	object       *ObjectAttrs
+	notification *Notification
+	hmacKey      *HMACKey
+}
+
+func (fs *resources) populate(ctx context.Context, c *Client, resource storage_v1_tests.Resource) error {
+	switch resource {
+	case storage_v1_tests.Resource_BUCKET:
+		bkt := c.Bucket(bucketIDs.New())
+		if err := bkt.Create(ctx, projectID, &BucketAttrs{}); err != nil {
+			return fmt.Errorf("creating bucket: %v", err)
+		}
+		attrs, err := bkt.Attrs(ctx)
+		if err != nil {
+			return fmt.Errorf("getting bucket attrs: %v", err)
+		}
+		fs.bucket = attrs
+	case storage_v1_tests.Resource_OBJECT:
+		// Assumes bucket has been populated first.
+		obj := c.Bucket(fs.bucket.Name).Object(objectIDs.New())
+		w := obj.NewWriter(ctx)
+		if _, err := w.Write([]byte("abcdef")); err != nil {
```

Comment:
Nit: As there are multiple places in the conf tests that will write to an object, is there a way instead of hardcoding a string, use a constant or variable instead?